Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -323,62 +325,72 @@ public void stopWebApp(String sid, String appid) throws IOException {
+            webAppList.addAll(this.listWebAppsBySidAndOS(sub.subscriptionId(), subscriptionIdToWebAppsOnWindowsMap,
+                force, OperatingSystem.WINDOWS));
         }
         return webAppList;
     }
 
     /**
-     * List all the Web Apps in selected subscriptions.
+     * List all the Web Apps on Linux in selected subscriptions.
+     *
+     * @param force flag indicating whether force to fetch most updated data from server
+     * @return list of Web App on Linux
      */
-    public List<ResourceEx<WebApp>> listAllWebAppsOnWindows(boolean force) {
-        List<ResourceEx<WebApp>> webAppList = new ArrayList<>();
-        List<Subscription> subscriptions = AzureMvpModel.getInstance().getSelectedSubscriptions();
-        for (Subscription sub : subscriptions) {
-            webAppList.addAll(this.listWebAppsOnWindowsBySubscriptionId(sub.subscriptionId(), force));
+    public List<ResourceEx<WebApp>> listAllWebAppsOnLinux(boolean force) {
+        final List<ResourceEx<WebApp>> webApps = new ArrayList<>();
+        for (final Subscription sb: AzureMvpModel.getInstance().getSelectedSubscriptions()) {
+            webApps.addAll(this.listWebAppsBySidAndOS(sb.subscriptionId(), subscriptionIdToWebAppsOnLinuxMap,
+                force, OperatingSystem.LINUX));
         }
-        return webAppList;
+        return webApps;
     }
 
     /**
-     * List Web App on Linux by Subscription ID.
+     * List all the Web Apps in selected subscriptions.
      *
-     * @param sid   subscription Id
      * @param force flag indicating whether force to fetch most updated data from server
-     * @return list of Web App on Linux
+     * @return list of Web App
      */
-    public List<ResourceEx<WebApp>> listWebAppsOnLinuxBySubscriptionId(String sid, boolean force) {
-        if (!force && subscriptionIdToWebAppsOnLinuxMap.containsKey(sid)) {
-            return subscriptionIdToWebAppsOnLinuxMap.get(sid);
+    public List<ResourceEx<WebApp>> listAllWebApps(boolean force) {
+        final List<ResourceEx<WebApp>> webApps = new ArrayList<>();
+        for (final Subscription sb: AzureMvpModel.getInstance().getSelectedSubscriptions()) {
+            webApps.addAll(this.listWebAppsBySidAndOS(sb.subscriptionId(), subscriptionIdToAllWebApps, force, null));
         }
-        List<ResourceEx<WebApp>> webAppList = new ArrayList<>();
+        return webApps;
+    }
+
+    /**
+     * List the Web Apps by subscription id and operating system.
+     * If os is null, get all the web apps no matter what kind of os is.
+     * @param sid        subscription id
+     * @param force      force flag indicating whether force to fetch most updated data from server
+     * @param os         operating system
+     * @return           list of Web App
+     */
+    public List<ResourceEx<WebApp>> listWebAppsBySidAndOS(final String sid,
+                                                          @NotNull final Map<String, List<ResourceEx<WebApp>>> cacheMap,
```

Comment:
I don't think `cacheMap` should be here in a public method. User of this API are not supposed to maintain a cachemap by themselves.